docs/typescript.md

```diff
@@ -1,14 +1,644 @@
+      return;
+    }
+    ...
+}
+```
+
+##### `@ts-expect-error` may be acceptable to use in tests, to intentionally break features
+
+###### Example (e299e95d-1c41-4251-85b6-f8064b22f577)
+
+✅
+
+```typescript
+// @ts-expect-error Suppressing to test runtime error handling
+
+// @ts-expect-error Intentionally testing invalid state
+
+// @ts-expect-error We are intentionally passing bad input.
+```
+
+##### If accompanied by a TODO comment, `@ts-expect-error` is acceptable to use for marking errors that have clear plans of being resolved
+
+<!-- TODO: Add example -->
+
+#### Avoid `any`
+
+`any` is the most dangerous form of explicit type declaration, and should be completely avoided.
+
+Unfortunately, `any` is also such a tempting escape hatch from the TypeScript type system that there's a strong incentive to use it whenever a nontrivial typing issue is encountered. Entire teams can easily be enticed into a pattern of unblocking feature development by introducing `any` with the intention of fixing it later. This is a major source of tech debt, and its destructive influence on the type safety of a codebase cannot be understated.
+
+Therefore, to prevent new `any` instances from being introduced into our codebase, it is not enough to rely on the `@typescript-eslint/no-explicit-any` ESLint rule. It's also necessary for all contributors to understand exactly why `any` is dangerous, and how it can be avoided.
+
+The key thing to remember about `any` is that it does not resolve errors, but only hides them. The errors still affect the code, while `any` makes it impossible to assess and counteract their influence.
+
+- `any` doesn't represent the widest type, or indeed any type at all. `any` is a compiler directive for _disabling_ static type checking for the value or type to which it's assigned.
+- `any` suppresses all error messages about its assignee. This makes code with `any` usage brittle against changes, since the compiler is unable to update its feedback even when the code has changed enough to alter or remove the error, or even add new type errors.
+- `any` subsumes all other types it comes into contact with. Any type that is in a union, intersection, is a property of, or has any other relationship with an `any` type or value becomes an `any` type itself. This represents an unmitigated loss of type information.
+
+###### Example (1fb5b0ad-61a9-4ad8-9d84-e29b78d88325)
+
+```typescript
+// Type of 'payload_0': 'any'
+const handler:
+  | ((payload_0: ComposableControllerState, payload_1: Patch[]) => void)
+  | ((payload_0: any, payload_1: Patch[]) => void);
+
+function returnsAny(): any {
+  return { a: 1, b: true, c: 'c' };
+}
+// Types of a, b, c are all `any`
+const { a, b, c } = returnsAny();
+```
+
+- `any` infects all surrounding and downstream code with its directive to suppress errors. This is the most dangerous characteristic of `any`, as it causes the encroachment of unsafe code with no guarantees about type safety or runtime behavior.
```

Comment:
Nit: Should this be a bullet point, if the next paragraph is not? Or should the next paragraph also be a bullet point?

Reply:
This line is part of the list in lines 457-459, with the example above only applying to line 459. Maybe a better example format with inline-html might make lists that include examples easier to parse? Otherwise, I'm not sure how to handle examples that are inserted into lists or otherwise don't quite fall under a dedicated entry.